src/crypto/crypto_sig.cc

```diff
@@ -847,7 +847,7 @@ bool SignTraits::DeriveBits(
       }
       break;
     case SignConfiguration::kVerify:
-      CHECK_EQ(params.key->GetKeyType(), kKeyTypePublic);
+      CHECK(params.key->GetKeyType() != kKeyTypeSecret);
```

Comment:
```suggestion
      CHECK_NE(params.key->GetKeyType(), kKeyTypeSecret);
```
Should we use this assertion instead?

Reply:
✔️